src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1341,7 +1342,49 @@ forSemanticModel.Syntax is { } semanticModelSyntax &&
                                 codeCoverageSpans,
                                 entryPointOpt: null);
 
-                            _moduleBeingBuiltOpt.SetMethodBody(methodSymbol.PartialDefinitionPart ?? methodSymbol, emittedBody);
+                            _moduleBeingBuiltOpt.SetMethodBody(GetSymbolForEmittedBody(methodSymbol), emittedBody);
+
+                            if (methodSymbol.ContainingType.IsExtension)
+                            {
+                                ILBuilder builder = new ILBuilder(_moduleBeingBuiltOpt, new LocalSlotManager(slotAllocator: null), OptimizationLevel.Release, areLocalsZeroed: false);
+
+                                // throw null;
```

Comment:
nit: consider expanding the comment to provide more context "// Emit methods in extensions as skeletons:"